packages/cli/generators/relation/utils.generator.js

```diff
@@ -0,0 +1,235 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const ast = require('ts-morph');
+const path = require('path');
+const tsquery = require('../../lib/ast-helper');
+const utils = require('../../lib/utils');
+
+exports.relationType = {
+  belongsTo: 'belongsTo',
+  hasMany: 'hasMany',
+};
+
+class AstLoopBackProject extends ast.Project {
+  constructor() {
+    super({
+      manipulationSettings: {
+        indentationText: ast.IndentationText.TwoSpaces,
+        insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces: false,
+        newLineKind: ast.NewLineKind.LineFeed,
+        quoteKind: ast.QuoteKind.Single,
+      },
+    });
+  }
+}
+exports.AstLoopBackProject = AstLoopBackProject;
+
+exports.getModelPrimaryKeyProperty = async function(fs, modelDir, modelName) {
+  const modelFile = path.join(modelDir, utils.getModelFileName(modelName));
+
+  const fileContent = await fs.read(modelFile, {});
+  return tsquery.getIdFromModel(fileContent);
+};
+
+exports.getModelPropertyType = function(modelDir, modelName, propertyName) {
+  const project = new this.AstLoopBackProject();
+
+  const modelFile = path.join(modelDir, utils.getModelFileName(modelName));
+  const sf = project.addExistingSourceFile(modelFile);
+  const co = this.getClassObj(sf, modelName);
+  return this.getPropertyType(co, propertyName);
+};
+
+exports.addFileToProject = function(project, dir, modelName) {
+  const fileName = path.resolve(dir, utils.getModelFileName(modelName));
+  return project.addExistingSourceFile(fileName);
+};
+
+exports.getClassObj = function(fileName, modelName) {
+  return fileName.getClassOrThrow(modelName);
+};
+
+exports.getClassConstructor = function(classObj) {
+  return classObj.getConstructors()[0];
+};
+
+exports.addExportController = async function(
+  generator,
+  fileName,
+  controllerClassName,
+  controllerFileName,
+) {
+  const project = new this.AstLoopBackProject();
+  let pFile;
+  const exportDeclaration = {
+    moduleSpecifier: './' + controllerFileName,
+  };
+
+  if (generator.fs.exists(fileName)) {
+    pFile = project.addExistingSourceFile(fileName);
```

Comment:
Based on the code coverage, this branch is never executed. Please add a test to verify the scenario implemented by this code.

IIUC, this is handling the case when the target file already exists, so the solution is to create a test that will add two controllers? 

I believe `lb4 controller` is adding controller exports too, please ensure that both generators are using the code for adding controller exports.

Reply:
added tests